src/CardDeck.php

```diff
@@ -293,9 +293,9 @@ protected function findCard(Model $model)
             }
         }
 
-        if (in_array($model->getId(), array_map($mapResults, $model->export([$model->idField])), true)) {
+        if (in_array($entity->getId(), array_map($mapResults, $entity->getModel()->export([$entity->idField])), true)) {
```

Comment:
I know you are proposing a bug fix and it looks right. But exporting all model rows in order to find out if an ID is present is horrible!

Reply:
findCard was coded like that before, so if you want to change the logic, please propose how. Have to be focused on fixing broken things due to the changes in atk4 due to time-constraints

Reply:
done